public/app/features/alerting/unified/api/featureDiscoveryApi.ts

```diff
@@ -25,37 +40,63 @@ export const featureDiscoveryApi = alertingApi.injectEndpoints({
       },
     }),
 
-    discoverDsFeatures: build.query<{ rulerConfig?: RulerDataSourceConfig }, { rulesSourceName: string }>({
-      queryFn: async ({ rulesSourceName }) => {
-        if (isGrafanaRulesSource(rulesSourceName)) {
-          return { data: { rulerConfig: GRAFANA_RULER_CONFIG } };
+    discoverDsFeatures: build.query<RulesSourceFeatures, { rulesSourceName: string } | { uid: string }>({
+      queryFn: async (rulesSourceIdentifier) => {
+        const dataSourceUID = getDataSourceUID(rulesSourceIdentifier);
+        if (!dataSourceUID) {
+          return { error: new Error(`Unable to find data source for ${rulesSourceIdentifier}`) };
+        }
+
+        if (isGrafanaRulesSource(dataSourceUID)) {
+          return {
+            data: {
+              name: GRAFANA_RULES_SOURCE_NAME,
+              uid: GRAFANA_RULES_SOURCE_NAME,
+              application: 'grafana',
+              rulerConfig: GRAFANA_RULER_CONFIG,
+            } satisfies RulesSourceFeatures,
+          };
         }
 
-        const dsSettings = getRulesDataSource(rulesSourceName);
-        if (!dsSettings) {
-          return { error: new Error(`Missing data source configuration for ${rulesSourceName}`) };
+        const dataSourceSettings = dataSourceUID ? getRulesDataSourceByUID(dataSourceUID) : undefined;
+        if (!dataSourceSettings) {
+          return { error: new Error(`Missing data source configuration for ${rulesSourceIdentifier}`) };
         }
 
-        const discoverFeaturesWithLogging = withPerformanceLogging(
-          'unifiedalerting/featureDiscoveryApi/discoverDsFeatures',
-          discoverFeatures,
-          {
-            dataSourceName: rulesSourceName,
-            endpoint: 'unifiedalerting/featureDiscoveryApi/discoverDsFeatures',
-          }
-        );
-
-        const dsFeatures = await discoverFeaturesWithLogging(dsSettings.name);
-
-        const rulerConfig: RulerDataSourceConfig | undefined = dsFeatures.features.rulerApiEnabled
-          ? {
-              dataSourceName: dsSettings.name,
-              apiVersion: dsFeatures.application === PromApplication.Cortex ? 'legacy' : 'config',
-            }
+        const features = await discoverFeaturesByUid(dataSourceSettings.uid);
+
+        const rulerConfig = features.features.rulerApiEnabled
+          ? ({
+              dataSourceName: dataSourceSettings.name,
+              apiVersion: features.application === PromApplication.Mimir ? 'config' : 'legacy',
```

Comment:
A bit worried about this change; I know we talked about it then but looking back at it – doesn't the prior version make more sense? 🤔 Did we change this because Cortex is now also reported as "Mimir"?

Reply:
Not much changed here actually, mostly whitespaces. what change do you mind?

Reply:
We've swapped

`features.application === PromApplication.Cortex ? 'legacy : 'config'` 
for 
`features.application === PromApplication.Mimir ? 'config' : 'legacy'` here which _seems_ the same but might be subtly different?

Reply:
Ah, I missed that! So in general, Mimir requires `config` version whereas Cortex requires `legacy`. I'm not sure about other flavours though. Thanos doesn't seem to implement Ruler API

Reply:
I have the same concern about changing the logic here. Previously, only `Cortex` was set to `legacy`, while everything else defaulted to `config`. With the new change, we're setting only the case `Mimir` to `config` and marking everything else as `legacy`. Is this the correct logic?

Reply:
We're reverting this to prevent any weird regressions